tutorials/SingleQubitGates/SingleQubitGates.ipynb

```diff
@@ -0,0 +1,874 @@
+    "        <td style=\"text-align:center; border:1px solid\">$\\langle\\psi| = \\overline{\\alpha}\\langle0| + \\overline{\\beta}\\langle1|$</td>\n",
+    "    </tr>\n",
+    "</table>\n",
+    "\n",
+    "Kets and bras give us a neat way to express [inner](../LinearAlgebra/LinearAlgebra.ipynb#Inner-Product) and [outer](../LinearAlgebra/LinearAlgebra.ipynb#Outer-Product) products. The inner product of $|\\phi\\rangle$ and $|\\psi\\rangle$ is $\\langle\\phi|\\psi\\rangle$, and their outer product is $|\\phi\\rangle\\langle\\psi|$. Notice that the norm of $|\\psi\\rangle$ is $\\sqrt{\\langle\\psi|\\psi\\rangle}$.\n",
+    "\n",
+    "This brings us to representing matrices. Recall that the outer product of two vectors of the same size produces a square matrix. We can use the sum of several outer products of simple vectors (with scalar factors) to express any matrix. For example, the $X$ gate can be expressed as follows:\n",
+    "\n",
+    "$$X = |0\\rangle\\langle1| + |1\\rangle\\langle0| \\\\\n",
+    "|0\\rangle\\langle1| + |1\\rangle\\langle0| =\n",
+    "\\begin{bmatrix} 1 \\\\ 0 \\end{bmatrix}\\begin{bmatrix} 0 & 1 \\end{bmatrix} +\n",
+    "\\begin{bmatrix} 0 \\\\ 1 \\end{bmatrix}\\begin{bmatrix} 1 & 0 \\end{bmatrix} =\n",
+    "\\begin{bmatrix} 0 & 1 \\\\ 0 & 0 \\end{bmatrix} + \\begin{bmatrix} 0 & 0 \\\\ 1 & 0 \\end{bmatrix} =\n",
+    "\\begin{bmatrix} 0 & 1 \\\\ 1 & 0 \\end{bmatrix}$$\n",
+    "\n",
+    "This can also be used for calculations:\n",
+    "\n",
+    "$$X|0\\rangle = \\big(|0\\rangle\\langle1| + |1\\rangle\\langle0|\\big)|0\\rangle = |0\\rangle\\langle1|0\\rangle + |1\\rangle\\langle0|0\\rangle = |0\\rangle\\big(\\langle1|0\\rangle\\big) + |1\\rangle\\big(\\langle0|0\\rangle\\big) = |0\\rangle(0) + |1\\rangle(1) = |1\\rangle$$\n",
+    "\n",
+    "> That last step may seem a bit confusing. Recall that $|0\\rangle$ and $|1\\rangle$ form an **orthonormal basis**. That is, they are both normalized, and they are orthogonal to each other.\n",
+    ">\n",
+    "> A vector is normalized if its norm is equal to $1$, which only happens if its inner product with itself is equal to $1$. This means that $\\langle0|0\\rangle = \\langle1|1\\rangle = 1$\n",
+    ">\n",
+    "> Two vectors are orthogonal to each other if their inner product equals $0$. This means that $\\langle0|1\\rangle = \\langle 1|0\\rangle = 0$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> ## Ket-bra decomposition\n",
+    ">\n",
+    "> This section deals with finding the ket-bra decompositions of quantum gates. This section is not necessary to start working with quantum gates, so feel free to skip it for now, and come back to it later.\n",
+    ">\n",
+    "> You can use the properties of [eigenvalues and eigenvectors](../LinearAlgebra/LinearAlgebra.ipynb#Eigenvalues-and-Eigenvectors) to find the ket-bra decomposition of any gate. Given a gate $A$, and its eigenvectors $|\\phi\\rangle$ and $|\\psi\\rangle$, if:\n",
```

Comment:
If I recall correctly, eigenvectors have to form an orthonormal basis for this to work? If that is so, let's mention this. I'd also use eigenvalues that are connected to the eigenvectors in their notation, for example, $x_\psi$ to match |𝜙⟩ - it will be easier on the reader.